docs/azure/sdk/unit-testing-mocking.md

```diff
@@ -231,6 +231,36 @@ var finder = new AboutToExpireSecretFinder(TimeSpan.FromDays(2), secretClient);
 
 This approach is useful when you would like to consolidate the dependency creation and share the client between multiple consuming classes.
 
+## Understand Azure Resource Manager clients
+
+In Azure Resource Manager libraries, the clients were designed to emphasize their relationship to one another, mirroring the service hierarchy. To achieve that goal, extension methods are widely used to add additional features to clients.
+
+For example, an Azure virtual machine resizes in an Azure resource group. The `Azure.ResourceManager.Compute` namespace models Azure virtual machine as `VirtualMachineResource` and `Azure.ResourceManager` namespace models Azure resource group as `ResourceGroupResource`.  To query the virtual machines for a resource group, you would write:
```

Comment:
In this 1st sentence, is "resizes" supposed to be "resource"?

Reply:
Yeah this should be a typo. Fixed